<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<meta name="description" content="Omics data (e.g. transcriptomics, proteomics,
    metagenomics...)  offer a detailed and multi-dimensional perspective
    on the molecular components and interactions within complex biological
    (eco)systems.  Analyzing these data requires adapted procedures, which
    are implemented as steps according to the recipes package.">
<title>Extra Recipes Steps for Dealing with Omics Data • scimo</title>
<script src="deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet">
<script src="deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- bootstrap-toc --><script src="https://cdn.jsdelivr.net/gh/afeld/bootstrap-toc@v1.0.1/dist/bootstrap-toc.min.js" integrity="sha256-4veVQbu7//Lk5TSmc7YV48MxtMy98e26cf5MrgZYnwo=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- search --><script src="https://cdnjs.cloudflare.com/ajax/libs/fuse.js/6.4.6/fuse.js" integrity="sha512-zv6Ywkjyktsohkbp9bb45V6tEMoWhzFzXis+LrMehmJZZSys19Yxf1dopHx7WzIKxr5tK2dVcYmaCk2uqdjF4A==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/autocomplete.js/0.38.0/autocomplete.jquery.min.js" integrity="sha512-GU9ayf+66Xx2TmpxqJpliWbT5PiGYxpaG8rfnBEk1LL8l1KGkRShhngwdXK1UgqhAzWpZHSiYPc09/NwDQIGyg==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/mark.min.js" integrity="sha512-5CYOlHXGh6QpOFA/TeTylKLWfB3ftPsde7AnmhuitiTX4K5SqCLBeKro6sPS8ilsz1Q4NRx3v8Ko2IBiszzdww==" crossorigin="anonymous"></script><!-- pkgdown --><script src="pkgdown.js"></script><meta property="og:title" content="Extra Recipes Steps for Dealing with Omics Data">
<meta property="og:description" content="Omics data (e.g. transcriptomics, proteomics,
    metagenomics...)  offer a detailed and multi-dimensional perspective
    on the molecular components and interactions within complex biological
    (eco)systems.  Analyzing these data requires adapted procedures, which
    are implemented as steps according to the recipes package.">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>
    

    <nav class="navbar fixed-top navbar-light navbar-expand-lg bg-light"><div class="container">
    
    <a class="navbar-brand me-2" href="index.html">scimo</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">0.0.1</small>

    
    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="nav-item">
  <a class="nav-link" href="reference/index.html">Reference</a>
</li>
<li class="nav-item dropdown">
  <a href="#" class="nav-link dropdown-toggle" data-bs-toggle="dropdown" role="button" aria-expanded="false" aria-haspopup="true" id="dropdown-articles">Articles</a>
  <div class="dropdown-menu" aria-labelledby="dropdown-articles">
    <a class="dropdown-item" href="articles/pedcan.html">Pediatric cancers</a>
  </div>
</li>
<li class="nav-item">
  <a class="nav-link" href="news/index.html">Changelog</a>
</li>
      </ul>
<form class="form-inline my-2 my-lg-0" role="search">
        <input type="search" class="form-control me-sm-2" aria-label="Toggle navigation" name="search-input" data-search-index="search.json" id="search-input" placeholder="Search for" autocomplete="off">
</form>

      <ul class="navbar-nav">
<li class="nav-item">
  <a class="external-link nav-link" href="https://github.com/abichat/scimo/" aria-label="github">
    <span class="fab fa fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>

    
  </div>
</nav><div class="container template-home">
<div class="row">
  <main id="main" class="col-md-9"><div class="section level1">
<div class="page-header"><h1 id="scimo">scimo<a class="anchor" aria-label="anchor" href="#scimo"></a>
</h1></div>
<!-- badges: start -->

<p><strong>scimo</strong> provides extra recipes steps for dealing with omics data, while also being adaptable to other data types.</p>
<div class="section level2">
<h2 id="installation">Installation<a class="anchor" aria-label="anchor" href="#installation"></a>
</h2>
<p>You can install <strong>scimo</strong> from GitHub with:</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># install.packages("remotes")</span></span>
<span><span class="fu">remotes</span><span class="fu">::</span><span class="fu">install_github</span><span class="op">(</span><span class="st">"abichat/scimo"</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="example">Example<a class="anchor" aria-label="anchor" href="#example"></a>
</h2>
<p>The <code>cheese_abundance</code> dataset describes fungal community abundance of 74 Amplicon Sequences Variants (ASVs) sampled from the surface of three different French cheeses.</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/abichat/scimo" class="external-link">scimo</a></span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="st">"cheese_abundance"</span>, <span class="st">"cheese_taxonomy"</span><span class="op">)</span></span>
<span></span>
<span><span class="va">cheese_abundance</span></span>
<span><span class="co">#&gt; # A tibble: 9 × 77</span></span>
<span><span class="co">#&gt;   sample    cheese    rind_type asv_01 asv_02 asv_03 asv_04 asv_05 asv_06 asv_07 asv_08 asv_09 asv_10 asv_11 asv_12 asv_13 asv_14 asv_15 asv_16</span></span>
<span><span class="co">#&gt;   &lt;chr&gt;     &lt;chr&gt;     &lt;chr&gt;      &lt;dbl&gt;  &lt;dbl&gt;  &lt;dbl&gt;  &lt;dbl&gt;  &lt;dbl&gt;  &lt;dbl&gt;  &lt;dbl&gt;  &lt;dbl&gt;  &lt;dbl&gt;  &lt;dbl&gt;  &lt;dbl&gt;  &lt;dbl&gt;  &lt;dbl&gt;  &lt;dbl&gt;  &lt;dbl&gt;  &lt;dbl&gt;</span></span>
<span><span class="co">#&gt; 1 sample1-1 Saint-Ne… Natural        1      0     38     40      1      2     31      8     15  20076    160     92     64     24     51      0</span></span>
<span><span class="co">#&gt; 2 sample1-2 Saint-Ne… Natural        3      4     38     61      4      4     48     14     20  32101    403    143    165     39    104      1</span></span>
<span><span class="co">#&gt; 3 sample1-3 Saint-Ne… Natural       28     16     33     23     31     29     21      1      7  12921    134     53     55     16     45      2</span></span>
<span><span class="co">#&gt; 4 sample2-1 Livarot   Washed         0      2      1      0      5      1      0      0      0   7823      2      0      0     42      0      2</span></span>
<span><span class="co">#&gt; 5 sample2-2 Livarot   Washed         0      0      4      0      1      1      2      0      0   6740      4      1      0     45      0      1</span></span>
<span><span class="co">#&gt; 6 sample2-3 Livarot   Washed         0      1      2      0      2      1      0      0      0   7484      6      1      0     43      0      7</span></span>
<span><span class="co">#&gt; 7 sample3-1 Epoisses  Washed         4      2      3      0      2      5      0      0      0   2486      1      1      1     23      0     24</span></span>
<span><span class="co">#&gt; 8 sample3-2 Epoisses  Washed         0      0      0      0      0      0      0      0      0   3686      2      0      0     28      0     54</span></span>
<span><span class="co">#&gt; 9 sample3-3 Epoisses  Washed         0      0      1      0      0      0      2      0      0   2988      2      1      0     22      0     36</span></span>
<span><span class="co">#&gt; # ℹ 58 more variables: asv_17 &lt;dbl&gt;, asv_18 &lt;dbl&gt;, asv_19 &lt;dbl&gt;, asv_20 &lt;dbl&gt;, asv_21 &lt;dbl&gt;, asv_22 &lt;dbl&gt;, asv_23 &lt;dbl&gt;, asv_24 &lt;dbl&gt;,</span></span>
<span><span class="co">#&gt; #   asv_25 &lt;dbl&gt;, asv_26 &lt;dbl&gt;, asv_27 &lt;dbl&gt;, asv_28 &lt;dbl&gt;, asv_29 &lt;dbl&gt;, asv_30 &lt;dbl&gt;, asv_31 &lt;dbl&gt;, asv_32 &lt;dbl&gt;, asv_33 &lt;dbl&gt;,</span></span>
<span><span class="co">#&gt; #   asv_34 &lt;dbl&gt;, asv_35 &lt;dbl&gt;, asv_36 &lt;dbl&gt;, asv_37 &lt;dbl&gt;, asv_38 &lt;dbl&gt;, asv_39 &lt;dbl&gt;, asv_40 &lt;dbl&gt;, asv_41 &lt;dbl&gt;, asv_42 &lt;dbl&gt;,</span></span>
<span><span class="co">#&gt; #   asv_43 &lt;dbl&gt;, asv_44 &lt;dbl&gt;, asv_45 &lt;dbl&gt;, asv_46 &lt;dbl&gt;, asv_47 &lt;dbl&gt;, asv_48 &lt;dbl&gt;, asv_49 &lt;dbl&gt;, asv_50 &lt;dbl&gt;, asv_51 &lt;dbl&gt;,</span></span>
<span><span class="co">#&gt; #   asv_52 &lt;dbl&gt;, asv_53 &lt;dbl&gt;, asv_54 &lt;dbl&gt;, asv_55 &lt;dbl&gt;, asv_56 &lt;dbl&gt;, asv_57 &lt;dbl&gt;, asv_58 &lt;dbl&gt;, asv_59 &lt;dbl&gt;, asv_60 &lt;dbl&gt;,</span></span>
<span><span class="co">#&gt; #   asv_61 &lt;dbl&gt;, asv_62 &lt;dbl&gt;, asv_63 &lt;dbl&gt;, asv_64 &lt;dbl&gt;, asv_65 &lt;dbl&gt;, asv_66 &lt;dbl&gt;, asv_67 &lt;dbl&gt;, asv_68 &lt;dbl&gt;, asv_69 &lt;dbl&gt;,</span></span>
<span><span class="co">#&gt; #   asv_70 &lt;dbl&gt;, asv_71 &lt;dbl&gt;, asv_72 &lt;dbl&gt;, asv_73 &lt;dbl&gt;, asv_74 &lt;dbl&gt;</span></span>
<span></span>
<span><span class="fu">glimpse</span><span class="op">(</span><span class="va">cheese_taxonomy</span><span class="op">)</span></span>
<span><span class="co">#&gt; Rows: 74</span></span>
<span><span class="co">#&gt; Columns: 9</span></span>
<span><span class="co">#&gt; $ asv     &lt;chr&gt; "asv_01", "asv_02", "asv_03", "asv_04", "asv_05", "asv_06", "asv_07", "asv_08", "asv_09", "asv_10", "asv_11", "asv_12", "asv_…</span></span>
<span><span class="co">#&gt; $ lineage &lt;chr&gt; "k__Fungi|p__Ascomycota|c__Dothideomycetes|o__Dothideales|f__Dothioraceae|g__Aureobasidium|s__Aureobasidium_Group_pullulans",…</span></span>
<span><span class="co">#&gt; $ kingdom &lt;chr&gt; "Fungi", "Fungi", "Fungi", "Fungi", "Fungi", "Fungi", "Fungi", "Fungi", "Fungi", "Fungi", "Fungi", "Fungi", "Fungi", "Fungi",…</span></span>
<span><span class="co">#&gt; $ phylum  &lt;chr&gt; "Ascomycota", "Ascomycota", "Ascomycota", "Ascomycota", "Ascomycota", "Ascomycota", "Ascomycota", "Ascomycota", "Ascomycota",…</span></span>
<span><span class="co">#&gt; $ class   &lt;chr&gt; "Dothideomycetes", "Eurotiomycetes", "Eurotiomycetes", "Eurotiomycetes", "Eurotiomycetes", "Eurotiomycetes", "Eurotiomycetes"…</span></span>
<span><span class="co">#&gt; $ order   &lt;chr&gt; "Dothideales", "Eurotiales", "Eurotiales", "Eurotiales", "Eurotiales", "Eurotiales", "Eurotiales", "Eurotiales", "Eurotiales"…</span></span>
<span><span class="co">#&gt; $ family  &lt;chr&gt; "Dothioraceae", "Aspergillaceae", "Aspergillaceae", "Aspergillaceae", "Aspergillaceae", "Aspergillaceae", "Aspergillaceae", "…</span></span>
<span><span class="co">#&gt; $ genus   &lt;chr&gt; "Aureobasidium", "Aspergillus", "Penicillium", "Penicillium", "Penicillium", "Penicillium", "Penicillium", "Penicillium", "Pe…</span></span>
<span><span class="co">#&gt; $ species &lt;chr&gt; "Aureobasidium Group pullulans", "Aspergillus fumigatus", "Penicillium Group camemberti caseifulvum fuscoglaucum commune", "P…</span></span></code></pre></div>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">list_family</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/split.html" class="external-link">split</a></span><span class="op">(</span><span class="va">cheese_taxonomy</span><span class="op">$</span><span class="va">asv</span>, <span class="va">cheese_taxonomy</span><span class="op">$</span><span class="va">family</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">list_family</span>, <span class="fl">2</span><span class="op">)</span></span>
<span><span class="co">#&gt; $Aspergillaceae</span></span>
<span><span class="co">#&gt; [1] "asv_02" "asv_03" "asv_04" "asv_05" "asv_06" "asv_07" "asv_08" "asv_09"</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; $Debaryomycetaceae</span></span>
<span><span class="co">#&gt;  [1] "asv_10" "asv_11" "asv_12" "asv_13" "asv_14" "asv_15" "asv_16" "asv_17" "asv_18" "asv_19" "asv_20" "asv_21" "asv_22"</span></span></code></pre></div>
<p>The following recipe will</p>
<ol style="list-style-type: decimal">
<li>aggregate the ASV variables at the family level, as defined by <code>list_family</code>;</li>
<li>transform counts into proportions;</li>
<li>discard variables those p-values are above 0.05 with a Kruskal-Wallis test against <code>cheese</code>.</li>
</ol>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">rec</span> <span class="op">&lt;-</span></span>
<span>  <span class="fu"><a href="https://recipes.tidymodels.org/reference/recipe.html" class="external-link">recipe</a></span><span class="op">(</span><span class="va">cheese</span> <span class="op">~</span> <span class="va">.</span>, data <span class="op">=</span> <span class="va">cheese_abundance</span><span class="op">)</span> <span class="op"><a href="reference/pipe.html">%&gt;%</a></span> </span>
<span>  <span class="fu"><a href="reference/step_aggregate_list.html">step_aggregate_list</a></span><span class="op">(</span><span class="fu"><a href="https://recipes.tidymodels.org/reference/has_role.html" class="external-link">all_numeric_predictors</a></span><span class="op">(</span><span class="op">)</span>,</span>
<span>                      list_agg <span class="op">=</span> <span class="va">list_family</span>, fun_agg <span class="op">=</span> <span class="va">sum</span><span class="op">)</span> <span class="op"><a href="reference/pipe.html">%&gt;%</a></span></span>
<span>  <span class="fu"><a href="reference/step_rownormalize_tss.html">step_rownormalize_tss</a></span><span class="op">(</span><span class="fu"><a href="https://recipes.tidymodels.org/reference/has_role.html" class="external-link">all_numeric_predictors</a></span><span class="op">(</span><span class="op">)</span><span class="op">)</span> <span class="op"><a href="reference/pipe.html">%&gt;%</a></span> </span>
<span>  <span class="fu"><a href="reference/step_select_kruskal.html">step_select_kruskal</a></span><span class="op">(</span><span class="fu"><a href="https://recipes.tidymodels.org/reference/has_role.html" class="external-link">all_numeric_predictors</a></span><span class="op">(</span><span class="op">)</span>, </span>
<span>                      outcome <span class="op">=</span> <span class="st">"cheese"</span>, cutoff <span class="op">=</span> <span class="fl">0.05</span><span class="op">)</span> <span class="op"><a href="reference/pipe.html">%&gt;%</a></span></span>
<span>  <span class="fu"><a href="https://recipes.tidymodels.org/reference/prep.html" class="external-link">prep</a></span><span class="op">(</span><span class="op">)</span></span>
<span></span>
<span><span class="va">rec</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; ── Recipe ─────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; ── Inputs</span></span>
<span><span class="co">#&gt; Number of variables by role</span></span>
<span><span class="co">#&gt; outcome:    1</span></span>
<span><span class="co">#&gt; predictor: 76</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; ── Training information</span></span>
<span><span class="co">#&gt; Training data contained 9 data points and no incomplete rows.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; ── Operations</span></span>
<span><span class="co">#&gt; • Aggregation of: asv_01, asv_02, asv_03, asv_04, asv_05, asv_06, asv_07, asv_08, asv_09, asv_10, asv_11, asv_12, asv_13, ... | Trained</span></span>
<span><span class="co">#&gt; • TSS normalization on: Aspergillaceae, Debaryomycetaceae, Dipodascaceae, Dothioraceae, Lichtheimiaceae, Metschnikowiaceae, ... | Trained</span></span>
<span><span class="co">#&gt; • Kruskal filtering against cheese on: Aspergillaceae, Debaryomycetaceae, Dipodascaceae, Dothioraceae, Lichtheimiaceae, ... | Trained</span></span>
<span></span>
<span><span class="fu"><a href="https://recipes.tidymodels.org/reference/juice.html" class="external-link">juice</a></span><span class="op">(</span><span class="va">rec</span><span class="op">)</span></span>
<span><span class="co">#&gt; # A tibble: 9 × 8</span></span>
<span><span class="co">#&gt;   sample    rind_type cheese         Debaryomycetaceae Dipodascaceae Saccharomycetaceae Saccharomycetales fam Incertae sedi…¹ Trichosporonaceae</span></span>
<span><span class="co">#&gt;   &lt;fct&gt;     &lt;fct&gt;     &lt;fct&gt;                      &lt;dbl&gt;         &lt;dbl&gt;              &lt;dbl&gt;                                 &lt;dbl&gt;             &lt;dbl&gt;</span></span>
<span><span class="co">#&gt; 1 sample1-1 Natural   Saint-Nectaire            0.719         0.0684           0.113                                 0.00130           0.000702</span></span>
<span><span class="co">#&gt; 2 sample1-2 Natural   Saint-Nectaire            0.715         0.0725           0.119                                 0.000801          0.000628</span></span>
<span><span class="co">#&gt; 3 sample1-3 Natural   Saint-Nectaire            0.547         0.277            0.0938                                0.000289          0.00239 </span></span>
<span><span class="co">#&gt; 4 sample2-1 Washed    Livarot                   0.153         0.845            0.000854                              0                 0.000349</span></span>
<span><span class="co">#&gt; 5 sample2-2 Washed    Livarot                   0.150         0.848            0.00106                               0                 0.000176</span></span>
<span><span class="co">#&gt; 6 sample2-3 Washed    Livarot                   0.160         0.837            0.00108                               0                 0.000212</span></span>
<span><span class="co">#&gt; 7 sample3-1 Washed    Epoisses                  0.0513        0.944            0.00327                               0                 0.000140</span></span>
<span><span class="co">#&gt; 8 sample3-2 Washed    Epoisses                  0.0558        0.941            0.00321                               0                 0.000176</span></span>
<span><span class="co">#&gt; 9 sample3-3 Washed    Epoisses                  0.0547        0.942            0.00329                               0                 0.000125</span></span>
<span><span class="co">#&gt; # ℹ abbreviated name: ¹​`Saccharomycetales fam Incertae sedis`</span></span></code></pre></div>
<p>To see which variables are kept and the associated p-values, you can use the <code>tidy</code> method on the third step:</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://generics.r-lib.org/reference/tidy.html" class="external-link">tidy</a></span><span class="op">(</span><span class="va">rec</span>, <span class="fl">3</span><span class="op">)</span></span>
<span><span class="co">#&gt; # A tibble: 13 × 4</span></span>
<span><span class="co">#&gt;    terms                                    pv kept  id                  </span></span>
<span><span class="co">#&gt;    &lt;chr&gt;                                 &lt;dbl&gt; &lt;lgl&gt; &lt;chr&gt;               </span></span>
<span><span class="co">#&gt;  1 Aspergillaceae                       0.0608 FALSE select_kruskal_WKayj</span></span>
<span><span class="co">#&gt;  2 Debaryomycetaceae                    0.0273 TRUE  select_kruskal_WKayj</span></span>
<span><span class="co">#&gt;  3 Dipodascaceae                        0.0273 TRUE  select_kruskal_WKayj</span></span>
<span><span class="co">#&gt;  4 Dothioraceae                         0.101  FALSE select_kruskal_WKayj</span></span>
<span><span class="co">#&gt;  5 Lichtheimiaceae                      0.276  FALSE select_kruskal_WKayj</span></span>
<span><span class="co">#&gt;  6 Metschnikowiaceae                    0.0509 FALSE select_kruskal_WKayj</span></span>
<span><span class="co">#&gt;  7 Mucoraceae                           0.0608 FALSE select_kruskal_WKayj</span></span>
<span><span class="co">#&gt;  8 Phaffomycetaceae                     0.0794 FALSE select_kruskal_WKayj</span></span>
<span><span class="co">#&gt;  9 Saccharomycetaceae                   0.0273 TRUE  select_kruskal_WKayj</span></span>
<span><span class="co">#&gt; 10 Saccharomycetales fam Incertae sedis 0.0221 TRUE  select_kruskal_WKayj</span></span>
<span><span class="co">#&gt; 11 Trichomonascaceae                    0.0625 FALSE select_kruskal_WKayj</span></span>
<span><span class="co">#&gt; 12 Trichosporonaceae                    0.0273 TRUE  select_kruskal_WKayj</span></span>
<span><span class="co">#&gt; 13 Wickerhamomyceteae                   0.177  FALSE select_kruskal_WKayj</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="notes">Notes<a class="anchor" aria-label="anchor" href="#notes"></a>
</h2>
<div class="section level3">
<h3 id="protection-stack-overflow-error">
<code>protection stack overflow</code> error<a class="anchor" aria-label="anchor" href="#protection-stack-overflow-error"></a>
</h3>
<p>If you have a very large dataset, you may encounter this error:</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="st">"pedcan_expression"</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://recipes.tidymodels.org/reference/recipe.html" class="external-link">recipe</a></span><span class="op">(</span><span class="va">disease</span> <span class="op">~</span> <span class="va">.</span>, data <span class="op">=</span> <span class="va">pedcan_expression</span><span class="op">)</span> <span class="op"><a href="reference/pipe.html">%&gt;%</a></span> </span>
<span>    <span class="fu"><a href="reference/step_select_cv.html">step_select_cv</a></span><span class="op">(</span><span class="fu"><a href="https://recipes.tidymodels.org/reference/has_role.html" class="external-link">all_numeric_predictors</a></span><span class="op">(</span><span class="op">)</span>, prop_kept <span class="op">=</span> <span class="fl">0.1</span><span class="op">)</span> </span>
<span><span class="co">#&gt; Error: protect(): protection stack overflow</span></span></code></pre></div>
<p>It is linked to <a href="https://github.com/tidymodels/recipes/issues/467" class="external-link">how <strong>R</strong> handles many variables in formulas</a>. To solve it, pass only the dataset to <code><a href="https://recipes.tidymodels.org/reference/recipe.html" class="external-link">recipe()</a></code> and manually update roles with <code><a href="https://recipes.tidymodels.org/reference/roles.html" class="external-link">update_role()</a></code>, like in the example below:</p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://recipes.tidymodels.org/reference/recipe.html" class="external-link">recipe</a></span><span class="op">(</span><span class="va">pedcan_expression</span><span class="op">)</span> <span class="op"><a href="reference/pipe.html">%&gt;%</a></span> </span>
<span>  <span class="fu"><a href="https://recipes.tidymodels.org/reference/roles.html" class="external-link">update_role</a></span><span class="op">(</span><span class="va">disease</span>, new_role <span class="op">=</span> <span class="st">"outcome"</span><span class="op">)</span> <span class="op"><a href="reference/pipe.html">%&gt;%</a></span> </span>
<span>  <span class="fu"><a href="https://recipes.tidymodels.org/reference/roles.html" class="external-link">update_role</a></span><span class="op">(</span><span class="op">-</span><span class="va">disease</span>, new_role <span class="op">=</span> <span class="st">"predictor"</span><span class="op">)</span> <span class="op"><a href="reference/pipe.html">%&gt;%</a></span> </span>
<span>  <span class="fu"><a href="reference/step_select_cv.html">step_select_cv</a></span><span class="op">(</span><span class="fu"><a href="https://recipes.tidymodels.org/reference/has_role.html" class="external-link">all_numeric_predictors</a></span><span class="op">(</span><span class="op">)</span>, prop_kept <span class="op">=</span> <span class="fl">0.1</span><span class="op">)</span> </span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; ── Recipe ─────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; ── Inputs</span></span>
<span><span class="co">#&gt; Number of variables by role</span></span>
<span><span class="co">#&gt; outcome:       1</span></span>
<span><span class="co">#&gt; predictor: 19196</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; ── Operations</span></span>
<span><span class="co">#&gt; • Top CV filtering on: all_numeric_predictors()</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="steps-for-variable-selection">Steps for variable selection<a class="anchor" aria-label="anchor" href="#steps-for-variable-selection"></a>
</h3>
<p>Like <a href="https://github.com/stevenpawley/colino" class="external-link"><strong>colino</strong></a>, <strong>scimo</strong> proposes 3 arguments for variable selection steps based on a statistic: <code>n_kept</code>, <code>prop_kept</code> and <code>cutoff</code>.</p>
<ul>
<li><p><code>n_kept</code> and <code>prop_kept</code> deal with how many variables will be kept in the preprocessed dataset, based on an exact count of variables or a proportion relative to the original dataset. They are mutually exclusive.</p></li>
<li><p><code>cutoff</code> removes variables whose statistic is below (or above, depending on the step) it. It could be used alone or in addition to the two others.</p></li>
</ul>
</div>
<div class="section level3">
<h3 id="dependencies">Dependencies<a class="anchor" aria-label="anchor" href="#dependencies"></a>
</h3>
<p><strong>scimo</strong> doesn’t introduce any additional dependencies compared to <strong>recipes</strong>.</p>
</div>
</div>
</div>
  </main><aside class="col-md-3"><div class="links">
<h2 data-toc-skip>Links</h2>
<ul class="list-unstyled">
<li><a href="https://cloud.r-project.org/package=scimo" class="external-link">View on CRAN</a></li>
<li><a href="https://github.com/abichat/scimo/" class="external-link">Browse source code</a></li>
<li><a href="https://github.com/abichat/scimo/issues" class="external-link">Report a bug</a></li>
</ul>
</div>

<div class="license">
<h2 data-toc-skip>License</h2>
<ul class="list-unstyled">
<li><a href="LICENSE.html">Full license</a></li>
<li><small>GPL (&gt;= 3)</small></li>
</ul>
</div>


<div class="citation">
<h2 data-toc-skip>Citation</h2>
<ul class="list-unstyled">
<li><a href="authors.html#citation">Citing scimo</a></li>
</ul>
</div>

<div class="developers">
<h2 data-toc-skip>Developers</h2>
<ul class="list-unstyled">
<li>Antoine BICHAT <br><small class="roles"> Author, maintainer </small> <a href="https://orcid.org/0000-0001-6599-7081" target="orcid.widget" aria-label="ORCID" class="external-link"><span class="fab fa-orcid orcid" aria-hidden="true"></span></a> </li>
<li><a href="authors.html">More about authors...</a></li>
</ul>
</div>

<div class="dev-status">
<h2 data-toc-skip>Dev status</h2>
<ul class="list-unstyled">
<li><a href="https://lifecycle.r-lib.org/articles/stages.html#experimental" class="external-link"><img src="https://img.shields.io/badge/lifecycle-experimental-orange.svg" alt="experimental"></a></li>
<li><a href="https://github.com/abichat/scimo/actions/workflows/R-CMD-check.yaml" class="external-link"><img src="https://github.com/abichat/scimo/actions/workflows/R-CMD-check.yaml/badge.svg" alt="R-CMD-check"></a></li>
</ul>
</div>

  </aside>
</div>


    <footer><div class="pkgdown-footer-left">
  <p></p>
<p>Developed by Antoine BICHAT.</p>
</div>

<div class="pkgdown-footer-right">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.7.</p>
</div>

    </footer>
</div>

  

  

  </body>
</html>
